src/transaction/internals.rs

```diff
@@ -48,6 +49,24 @@ impl Transaction {
     pub fn set_bytecode(&mut self, bytecode: Witness) -> Option<()> {
         self._set_bytecode(bytecode)
     }
+
+    pub fn inputs_mut(&mut self) -> &mut [Input] {
```

Comment:
any reason why these should be gated as internal? Seems useful if we wanted to have the ability to edit these fields without big match blocks around the tx in the VM or client for malleation.

Reply:
We should use the methods and API instead, that will (or at least should) guarantee tx consistency.

Exposing the internals is not only allowing the users to use the API to create invalid transactions, but will make their code more vulnerable to breaking changes. Ideally, somewhere in the future, we should never have to `match` these enums and use only the available API.

However, if for some reason the user still wants to manipulate the shielded attributes of a transaction, he can just, knowingly, activate this feature